### PR TITLE
clean up macro examples and README.md, remove run.dart script

### PR DESCRIPTION
**Pull request by jakemac53**
- Delete run.dart script and update the README.md to indicate you can just pass `--enable-experiment=macros`.
- Break up `user_main.dart` into separate scripts for each macro example, which allows you to run the ones that do work, and also keeps the example apps more targetted. These were only combined into one previously to make it easier to run them through `run.dart`.
- Update the observable example so it is runnable.
- Update the data class example to create the unnamed constructor instead of one called `gen`.

### CODE REVIEW

**Comment by mnordine** [comment]
I checked out this branch, and did:
```$ dart --enable-experiment=macros run bin/json_serializable_main.dart                                                           
Error: Exception when reading 'org-dartlang-macro:///a/b/c/.dart_tool/package_config.json': FileSystemException(uri=org-dartlang-macro:///a/b/c/.dart_tool/package_config.json; message=StandardFileSystem only supports file:* and data:* URIs)
Crash when compiling org-dartlang-macro:///a/b/c/main.dart:
Invalid argument(s): Exception when reading 'org-dartlang-macro:///a/b/c/.dart_tool/package_config.json': FileSystemException(uri=org-dartlang-macro:///a/b/c/.dart_tool/package_config.json; message=StandardFileSystem only supports file:* and data:* URIs)

#0      ProcessedOptions._findPackages.checkInDir (package:front_end/src/base/processed_options.dart:671:9)
<asynchronous suspension>
#1      ProcessedOptions._findPackages (package:front_end/src/base/processed_options.dart:676:22)
<asynchronous suspension>
#2      ProcessedOptions._getPackages (package:front_end/src/base/processed_options.dart:556:24)
<asynchronous suspension>
#3      ProcessedOptions.getUriTranslator (package:front_end/src/base/processed_options.dart:481:32)
<asynchronous suspension>
#4      generateKernelInternal.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:73:37)
<asynchronous suspension>
#5      withCrashReporting (package:front_end/src/fasta/crash.dart:133:12)
<asynchronous suspension>
#6      kernelForProgramInternal.<anonymous closure> (package:front_end/src/api_prototype/kernel_generator.dart:62:29)
<asynchronous suspension>
#7      CompilerContext.clear (package:front_end/src/fasta/compiler_context.dart:139:3)
<asynchronous suspension>
#8      kernelForProgramInternal (package:front_end/src/api_prototype/kernel_generator.dart:61:10)
<asynchronous suspension>
#9      _compileMacros (package:front_end/src/kernel_generator_impl.dart:349:7)
<asynchronous suspension>
#10     precompileMacros (package:front_end/src/kernel_generator_impl.dart:305:16)
<asynchronous suspension>
#11     generateKernelInternal.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:104:11)
<asynchronous suspension>
#12     withCrashReporting (package:front_end/src/fasta/crash.dart:133:12)
<asynchronous suspension>
#13     generateKernel.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:49:12)
<asynchronous suspension>
#14     CompilerContext.clear (package:front_end/src/fasta/compiler_context.dart:139:3)
<asynchronous suspension>
#15     generateKernel (package:front_end/src/kernel_generator_impl.dart:48:10)
<asynchronous suspension>
#16     kernelForModule (package:front_end/src/api_prototype/kernel_generator.dart:100:11)
<asynchronous suspension>
#17     SingleShotCompilerWrapper.compileInternal (file:///Volumes/Work/s/w/ir/x/w/sdk/pkg/vm/bin/kernel_service.dart:419:11)
<asynchronous suspension>
#18     Compiler.compile.<anonymous closure> (file:///Volumes/Work/s/w/ir/x/w/sdk/pkg/vm/bin/kernel_service.dart:225:45)
<asynchronous suspension>
#19     _processLoadRequest (file:///Volumes/Work/s/w/ir/x/w/sdk/pkg/vm/bin/kernel_service.dart:906:37)
<asynchronous suspension>

```
Similar result for all examples.

**Comment by jakemac53** [comment]
@mnordine did you run a `dart pub get`?

**Comment by mnordine** [comment]
> @mnordine did you run a dart pub get?

Yes

> Also fwiw, the observable one is I think the only one that works right this moment

Ok, there is no file in `bin` for that one.

**Comment by davidmorgan** [comment]
@mnordine that issue was fixed very on [Jan 19th](https://dart-review.googlesource.com/c/sdk/+/347340)--you'll need a more recent SDK.

**Comment by jakemac53** [comment]
> Ok, there is no file in `bin` for that one.

Oh sorry, I meant the `auto_dispose` one, in this PR. I have another PR out which adds a main for `observable` which does in fact work as well.

File: working/macros/example/bin/auto_dispose_main.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import 'package:macro_proposal/auto_dispose.dart';
```
Comment (davidmorgan) [inline working/macros/example/bin/auto_dispose_main.dart:1]: Copyright header?

File: working/macros/example/bin/data_class_main.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
```
Comment (davidmorgan) [inline working/macros/example/bin/data_class_main.dart:1]: 2024?

**Review by davidmorgan — APPROVED** [review]

**Pull request merged**